internal/proxy/task_upsert.go

```diff
@@ -348,19 +355,57 @@ func (it *upsertTask) PreExecute(ctx context.Context) error {
 		log.Warn("Fail to insertPreExecute", zap.Error(err))
 		return err
 	}
+	it.result.InsertCnt = int64(it.upsertMsg.InsertMsg.NumRows)
 
-	err = it.deletePreExecute(ctx)
+	// query to find if pk were exist
+	// if pk not exist, then skip delete operate
+	queryResult, err := it.queryPK()
```

Comment:
the queryPK has to be configurable.
This might affect the upsert performance because it has to be strong consistency